### PR TITLE
hotfix/login-validation-exception

### PR DESCRIPTION
**Pull request by ekaterina-demeshko-idf**

### CODE REVIEW

**Review by nikolay-ahramovich-idf — APPROVED** [review]

File: packages/domain/lib/usecase/check_user_usecase.dart [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+class CheckUserUseCase implements UseCaseParams<UserModel, Future<void>> {
   final RemoteDBRepository _firestoreRepository;
 
   CheckUserUseCase(this._firestoreRepository);
 
   @override
-  Future<bool> call(UserModel user) async =>
-      _firestoreRepository.checkUserExist(user);
+  Future<void> call(UserModel user) async {
+    if (!(await _firestoreRepository.checkUserExist(user))) {
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/usecase/check_user_usecase.dart]: I'd move await _firestoreRepository.checkUserExist(user) to separate var

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: packages/domain/lib/usecase/check_user_usecase.dart (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply packages/domain/lib/usecase/check_user_usecase.dart]: it is void method

Comment (ekaterina-demeshko-idf) [reply packages/domain/lib/usecase/check_user_usecase.dart]: а ой

File: packages/presentation/lib/screen/login/login_bloc.dart [hunk 2]
```diff
@@ -92,16 +93,16 @@ class _LoginBloc extends BlocImpl<LoginScreenArguments, LoginData>
       validationResponse?.passwordValidation,
     );
     if (_formKey.currentState?.validate() ?? false) {
-      bool isSuccess = await _checkUserUseCase(user);
-      if (!isSuccess) {
+      try {
+        await _checkUserUseCase(user);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/login/login_bloc.dart]: validation usecase should also be called in try block

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/screen/login/login_bloc.dart [hunk 3]
```diff
@@ -85,24 +86,24 @@ class _LoginBloc extends BlocImpl<LoginScreenArguments, LoginData>
...
-        _formKey.currentState?.validate();
-      } else {
+    try {
+      final LoginValidationError? validationResponse =
+          await _loginValidationUseCase(user);
+      validationModel = ValidationModel(
+        validationResponse?.emailValidation,
+        validationResponse?.passwordValidation,
+      );
+      if (_formKey.currentState?.validate() ?? false) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/login/login_bloc.dart]: still not good - u should call form.validate only in the catch block. it shouldn't be in try

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/domain/lib/usecase/login_validation_usecase.dart [hunk 4]
```diff
@@ -16,23 +16,23 @@ class LoginValidationUseCase
...
+        (element) => !element.isValid(params.password as String));
 
     final ValidationErrorType? passwordFailedType =
         getEnumByValidator(passwordFailed);
-
-    return LoginValidationError(
-      emailFailedType,
-      passwordFailedType,
-    );
+    if (emailFailedType != null && passwordFailedType != null) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/lib/usecase/login_validation_usecase.dart]: what if only email/password is incorrect ...

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**